### PR TITLE
feat(Page): Add global timeout setting

### PR DESCRIPTION
**Pull request by radu125**
References #1514

### CODE REVIEW

**Comment by radu125** [comment]
I signed the CLA.

**Comment by aslushnikov** [comment]
Can you please eliminate the default 30000 timeout from the NavigatorWatcher?

https://github.com/GoogleChrome/puppeteer/blob/8e9c54a789c9d51f9861053b21565ad21a5ffe6f/lib/NavigatorWatcher.js#L41-L48

I'd do this by passing timeout explicitly to the NavigatorWatcher as one of the arguments in its constructor.

**Comment by radu125** [comment]
I did it, but the AppVeyor build failed, and I can't figure out why that test failed.

File: test/test.js [hunk 1]
```diff
@@ -464,6 +464,14 @@ describe('Page', function() {
     });
   });
 
+  describe('Page.setDefaultNavigationTimeout', function() {
+    it('should work', async({page, server}) => {
```
Comment (aslushnikov) [inline test/test.js]: We generally try hard not to reach out to private fields.
Let's drop this test – i think the other one you add is sufficient.

**Review by aslushnikov — COMMENTED** [review]
@raduaron26 no worries about appveyour, some of our tests are flaky. 

Thank you for working on this. I have a minor comment, other than that it's good to go.

**Comment by radu125** [comment]
I removed the test. I'm glad I could contribute and thank you too for your help :)

**Review by aslushnikov — APPROVED** [review]

File: docs/api.md [hunk 2]
```diff
@@ -78,6 +78,7 @@
   * [page.select(selector, ...values)](#pageselectselector-values)
   * [page.setContent(html)](#pagesetcontenthtml)
   * [page.setCookie(...cookies)](#pagesetcookiecookies)
+  * [page.setDefaultNavigationTimeout(timeout)](#pagesetdefaultnavigationtimeouttimeout)
```
Comment (aslushnikov) [inline docs/api.md:81]: So this setting affects 5 methods: `page.goto`, `page.goBack`, `page.goForward`, `page.reload` and `page.waitForNavigation`.
Can you please:
- refer to the methods from the doc of the method, explaining that this method will change defaults for them, and that the default value is 30 seconds
- update documentation for the methods themselves, citing `page.setDefaultNavigationTimeout`.

This should make it more understandable for our clients.

**Review by aslushnikov — CHANGES_REQUESTED** [review]
@raduaron26 I've just figured there's one last nit for the documentation before this should be merged. Thanks again for taking your time for this

**Comment by radu125** [comment]
I updated the documentation, let me know if you want me to make any tweaks to what I wrote.

**Review by aslushnikov — APPROVED** [review]
Thank you, this looks good.

**Comment by stackflows** [comment]
@aslushnikov: This only affects navigation. waitForSelector() etc. should have a similar global "default" setting. Shall I write a feature request for this?

**Comment by Peta5000** [comment]
@stackflows Did you write a feature request for this? This feature would be so precious.

**Comment by spinningarrow** [comment]
@Peta5000 it looks like #1514 is the issue for this but it is closed. I've added a comment there voting to reopen.

UPDATE: This is a part of #2079

**Pull request merged**